app/controllers/PropertyLinkingController.scala

```diff
@@ -69,20 +70,23 @@ class PropertyLinkingController @Inject()(propertyLinksConnector: PropertyLinkin
     }
   }
 
-  private def detailed(authorisation: PropertiesView)(implicit hc: HeaderCarrier,
-                                                      memoParty: Memoize[APIParty, Future[Option[(APIParty, GroupAccount)]]]): Future[PropertyLink] = {
+  private def detailed(authorisation: PropertiesView)(implicit hc: HeaderCarrier, cache: GroupCache): Future[PropertyLink] = {
     for {
-      apiPartiesWithGroupAccounts <- getGroupAccounts(authorisation)
+      apiPartiesWithGroupAccounts <- withGroupAccounts(authorisation)
       parties = apiPartiesWithGroupAccounts.flatMap { case (p: APIParty, g: GroupAccount) => Party.fromAPIParty(p, g) }
     } yield PropertyLink.fromAPIAuthorisation(authorisation, parties)
   }
 
-  private def getGroupAccounts(authorisation: PropertiesView)
-                              (implicit hc: HeaderCarrier,
-                               memoParty: Memoize[APIParty, Future[Option[(APIParty, GroupAccount)]]]): Future[Seq[(APIParty, GroupAccount)]] = {
-    Future.traverse(authorisation.parties)(memoParty { party =>
-      groupAccountsConnector.get(party.authorisedPartyOrganisationId).map(_.map(groupAccount => (party, groupAccount)))
-    }).map(_.flatten)
+  private def withGroupAccounts(authorisation: PropertiesView)(implicit hc: HeaderCarrier, cache: GroupCache): Future[Seq[(APIParty, GroupAccount)]] = {
```

Comment:
Isn't the easier solution to retain the existing code and just cache the connector.get:

`Future.traverse(authorisation.parties)(party =>
      cache(groupAccountsConnector.get)(party.authorisedPartyOrganisationId).map(_.map(groupAccount => (party, groupAccount)))
).map(_.flatten)`

Only requires a single traversal of authorisation.parties.